### PR TITLE
Spilling your spaghetti

### PR DESCRIPTION
**Pull request by ghost**
[content][sprites]
Idea requested by someone named "tryepod", who couldn't code this due to hard drive issues.
Basically, if spaghetti is in pockets, 5% chance for it to fall out per tick, 10% if running, 20% if saying something, and all the time when knocked down, creating a cleanable decal mess on the floor.
When spaghetti is spilled, you stutter most of the time, (90%) or occasionally say nothing at all for a while. (10%)
Spilled spaghetti can also be vacuumed as a reagent, which causes similar effects to spilled spaghetti when consumed. (80% chance to stutter, 10% to mute and 10% to spill more spaghetti)

Next, I'll try work on a symptom that spawns it in pockets, but not until I find a way to code spawning items in pockets or get told how to do it.
:cl:
 * rscadd: Having spaghetti in your pockets will now cause you to spill it sometimes, bursting into treats not included.
 * rscadd: Spaghetti reagent, found in spilled spaghetti, with similar "effects".
 * rscadd: Spaghetti reagent is now applied during fevers.

### CODE REVIEW

**Comment by toomykins** [comment]
image

**Comment by RaccoonBro** [comment]
Make it spill when throwing it as well for some food fights

**Comment by Eneocho** [comment]
Make the spawning a secondary effect of the sweaty palms symptom so your palms are sweaty, knees weak, arms heavy...

**Comment by Eneocho** [comment]
Also, the Biolobulin Effect symptom spawns items in your hand, perhaps you can use some of it's code to spawn spaghetti on the pockets.
Pocketspawning code is on the outfit datums, I think, since many jobs spawn with things on the pocket at roundstart (like engineer spawning with the PDA on the left pocket since the belt slot has a toolbelt).

**Comment by Eneocho** [comment]
If you clean the spaghetti decal with the wet/dry vacuum, do you pick up nutriment? Spaghetti essence? Anything?
Make it pick up a spaghetti chem that can only be acquired this way which spills spaghetti if you consume it, regardless of whether or not you had spaghetti in your pocket. Perhaps it's the way you can drop the spaghetti as a disease. It gives you 1u of spaghetti essence every few seconds which makes you drop the spaghetti. It'd be a way to cause the spaghetti drop without spawning the item on the pocket.

File: code/game/objects/effects/decals/Cleanable/misc.dm [hunk 1]
```diff
@@ -201,6 +201,14 @@
 	icon = 'icons/effects/tomatodecal.dmi'
 	random_icon_states = list("smashed_pie")
 
+/obj/effect/decal/cleanable/spaghetti_spill
+	name = "spilled spaghetti"
+	desc = "A result of bursting into treats from awkward situations."
+	density = 0
+	anchored = 1
+	icon = 'icons/effects/tomatodecal.dmi'
```
Comment (Kurfursten) [inline code/game/objects/effects/decals/Cleanable/misc.dm:210]: maybe you should make `reagent = FLOUR` or `TOMATOJUICE`

Comment (ghost) [reply code/game/objects/effects/decals/Cleanable/misc.dm:210]: Oh okay

Comment (ghost) [reply code/game/objects/effects/decals/Cleanable/misc.dm:210]: Done

Comment (ghost) [reply code/game/objects/effects/decals/Cleanable/misc.dm:210]: Decided to go above and beyond

**Comment by Eneocho** [comment]
You can add a symptom to the palms sweaty stage that just gives you some spaghetti reagent every now and then.

**Comment by ghost** [comment]
> You can add a symptom to the palms sweaty stage that just gives you some spaghetti reagent every now and then.

That's in fever handling, not a symptom, also done.

**Comment by ghost** [comment]
> Also, the Biolobulin Effect symptom spawns items in your hand, perhaps you can use some of it's code to spawn spaghetti on the pockets.
> Pocketspawning code is on the outfit datums, I think, since many jobs spawn with things on the pocket at roundstart (like engineer spawning with the PDA on the left pocket since the belt slot has a toolbelt).

It's really not that easy, I'd have to entirely rewrite put_in_hands to make it work.

**Comment by Eneocho** [comment]
Requesting for thrown spaghetti to have a 50% chance of becoming spilled spaghetti too.

**Pull request merged**